### PR TITLE
added logic to add a user

### PR DESCRIPTION
**Pull request by nicole-reed**

### CODE REVIEW

File: src/controllers/user.controller.ts [hunk 1]
```diff
@@ -11,6 +11,18 @@ export const getUsers = async (): Promise<HttpResponse> => {
     };
 };
 
+export const addUser = async (request: unknown): Promise<HttpResponse> => {
+    const validatedRequest = addUserSchema.parse(request);
+    const name = validatedRequest.body.name;
+    const username = validatedRequest.body.username;
```
Comment (derekslarson) [inline src/controllers/user.controller.ts]: destructuring these might clean things up a bit

File: src/repositories/user.repository.ts [hunk 2]
```diff
@@ -13,6 +13,12 @@ export const getUsers = async (): Promise<User[]> => {
     return users;
 };
 
+export const addUser = async (name: string, username: string): Promise<void> => {
+    const client = new Firestore();
+
+    await client.collection("users").add({ name, username });
```
Comment (derekslarson) [inline src/repositories/user.repository.ts]: Does username need to be unique? Seems like it should be.

**Review by derekslarson — CHANGES_REQUESTED** [review]
Need clarity on `user.username`

File: src/repositories/user.repository.ts (hunk 2 above)
Comment (nicole-reed) [reply src/repositories/user.repository.ts]: @cstst fixed this!

File: src/models/user.ts [hunk 3]
```diff
@@ -1,7 +1,15 @@
 import { z } from "zod";
 
 export const userSchema = z.object({
-    name: z.string()
+    name: z.string(),
+    username: z.string()
+});
```
Comment (derekslarson) [inline src/models/user.ts]: needs to have an `id` property

**Review by derekslarson — CHANGES_REQUESTED** [review]
just one small thing

**Review by derekslarson — APPROVED** [review]

**Pull request merged**